### DISCUSSION TITLE
how to "refresh" a dev environment

### DISCUSSION THREAD

**Discussion by j1philli** [pkgxdev/pkgx#791] [Help] [answered]
Reference: https://github.com/orgs/pkgxdev/discussions/768
I enabled a dev environment where a dependency was not specific enough and installed python 3.12. The utility that required python was not compatible with 3.12. The team updated all the python dependencies in pantry to be more specific and released 1.0.1 of pkgx.
I expect that ```dev off``` then ```dev``` would refresh the environment and resolve dependencies to proper versions.
What is the expected line of action here to resolve my dependency versions?

**Comment by j1philli** [comment]
even after deleting both the tool requiring python and python itself from .pkgx, and cd . in the dev environment, pkgx is still fetching python 3.12 even though the dependency has been updated in the [pantry](https://github.com/pkgxdev/pantry/blob/fe4553718c175c234572c0eed70dabdbd3369e47/projects/python-poetry.org/package.yml#L9)

**Comment by mxcl** [comment]
Currently you have to `cd && cd -`, ie. change directory.

Sorry this sucks, polishing `dev` is on my near term TODO. I imagine we will eventually “watch” the keyfiles and update as required but initially I imagine it will be more explicit.

If your `dev` environment adds python and python 3.12 is installed that is what you will get unless you constrain it more youself. Eg any python project will result in this. The pantry has been constrained to avoid 3.12 but that is just what pkgs that use python are constrained to, this is a layer above that.

There is talk to automatically avoiding new releases of certain pkgs in such situations but we're not clear on if we should do that or not yet.

**Reply by j1philli** [reply, marked as answer]
it was because I did not have the latest pantry. https://github.com/orgs/pkgxdev/discussions/768#discussioncomment-7226186
Assumed this was pulled on a regular basis and I didn't have to do it manually
